pkg/river/parser/parser.go

```diff
@@ -30,6 +31,11 @@ func ParseExpression(expr string) (ast.Expr, error) {
 	p := newParser("", []byte(expr))
 
 	e := p.ParseExpression()
+
+	if p.tok != token.TERMINATOR && p.tok != token.EOF {
+		p.addErrorf("did not parse the expression in full - unexpected token of type %s", p.tok.String())
+	}
```

Comment:
I'm not totally sure on this one. I'd like to use `p.expect` because that's the standard way we assert a token. 

Expressions should _always_ end in token.TERMINATOR, and a token.TERMINATOR is expected to be injected prior to token.EOF (if it doesn't, that could be another bug that needs investigating).

I think it's safe to replace this entire section with `p.expect(token.TERMINATOR)`.